### PR TITLE
Setup optimization

### PR DESCRIPTION
**Pull request by spherel**

### CODE REVIEW

**Comment by lispc** [comment]
Better give some metric here? For example, old setup code takes XX seconds for degree 18, and new code takes YY seconds.

**Comment by spherel** [comment]
**[new]**
- test_parameter_serialization, k = 10, **8.06s**, 
- test_commit_lagrange, k = 10, **7.83s**

**[old]**
- test_parameter_serialization, k = 10, **9.99s**
- test_commit_lagrange, k = 10, **9.71s**

**[env]**
- number of thread = 8

**Comment by spherel** [comment]
<html>
<body>
<p>
Here is the profiling result for some larger k in the same testing environment:

</p>
<google-sheets-html-origin>

k | unsafe_setup (old) | unsafe_setup (new) | improvement | compute monomials (old) | compute monomials (new) | improvement
-- | -- | -- | -- | -- | -- | --
16 | 69.266 s | 56.859 s | 17.91% | 15.77 s | 3.6578 s | 76.81%
18 | 291.14 s | 244.02 s | 16.18% | 62.928 s | 14.965 s | 76.22%
20 | 1233.9 s | 1033.6 s | 16.23% | 252.47 s | 59.618 s | 76.39%


</body>
</html>

File: halo2_proofs/src/poly/commitment.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 //!
 //! [halo]: https://eprint.iacr.org/2019/1021
 
+use super::super::multicore;
```
Comment (CPerezz) [inline halo2_proofs/src/poly/commitment.rs:6]: Please, avoid `super::super` patterns. Start from `crate::`.

**Review by CPerezz — CHANGES_REQUESTED** [review]
Small typo. 

The rest looks good!

**Review by CPerezz — COMMENTED** [review]
Sorry, let me run the tests first. Then we can approve and merge

**Comment by spherel** [comment]
Great! Thanks!

**Pull request merged**